apps/antalmanac/src/components/RightPane/SectionTable/SectionTableBody.tsx

```diff
@@ -362,26 +385,129 @@ const SectionTableBody = withStyles(styles)((props: SectionTableBodyProps) => {
             setAddedCourse(doAdd);
         };
 
-        toggleHighlight();
-        AppStore.on('addedCoursesChange', toggleHighlight);
-        AppStore.on('currentScheduleIndexChange', toggleHighlight);
+        const checkScheduleConflict = () => {
+            if (AppStore.getEventsInCalendar().length < 1) {
+                setScheduleConflict(false);
+                return;
+            }
+
+            // An array of lists of time information on every added event
+            const calendarEventTimes = AppStore.getEventsInCalendar().map((event) => {
+                const courseDay = event.start.getDay();
+                const courseStartTime = event.start.toString().split(' ')[4];
+                const courseEndTime = event.end.toString().split(' ')[4];
+                return { day: courseDay, startTime: courseStartTime, endTime: courseEndTime };
+            });
+
+            const coursePaneEvent = {
+                // If there already exists a more well-written way to translate secton.meetings days (string) into a number, LMK
+                // Converts SuTuTh -> [Su, Tu, Th] -> [0, 2, 4]
+                day: section.meetings[0].days.match(/[A-Z][a-z]*/g)?.map((day: string) => translateDaysToNums[day]),
+                startTime: '',
+                endTime: '',
+            };
+
+            // Wasn't sure if running the function constantly in useEffect was bad, so moved it out just in case
+            const translatedTimeString = translateTimeString();
+            if (translatedTimeString) {
+                coursePaneEvent.startTime = translatedTimeString.startTime;
+                coursePaneEvent.endTime = translatedTimeString.endTime;
+            }
+
+            for (let i = 0; i < calendarEventTimes.length; i++) {
+                // Check if there is day overlap
+                if (!coursePaneEvent?.day?.includes(calendarEventTimes[i]?.day)) {
+                    continue;
+                }
+
+                // Then, IF the course ( starts AND ends BEFORE) OR ( starts AND ends AFTER), it's good
+                // Currently, it's coded to be greater/less than or equal to, but... if class A ends at 1:00 PM and class B starts at 1:00 PM, you're kinda boned, no?
+                const happensBefore =
+                    coursePaneEvent.startTime <= calendarEventTimes[i].startTime &&
+                    coursePaneEvent.endTime <= calendarEventTimes[i].startTime;
+                const happensAfter =
+                    coursePaneEvent.startTime >= calendarEventTimes[i].endTime &&
+                    coursePaneEvent.endTime >= calendarEventTimes[i].endTime;
+
+                // Otherwise, mark it in red
+                if (!(happensBefore || happensAfter)) {
+                    setScheduleConflict(true);
+                    return;
+                }
+            }
+            setScheduleConflict(false);
+            return;
+        };
+
+        const handleEventListeners = () => {
```

Comment:
More naming nitpicks for this and some other functions:
1. `handleEventListeners` is hella generic. Maybe `updateFormatting`?
2. `toggleHighlight` doesn't actually always toggle the highlight. Should be `updateHighlight`
3. `checkScheduleConflict` not only checks for the schedule conflict, but updates the overlay as well. Can't think of a good name off the top of my head, but it should convey all the actions that the function takes.

Reply:
checkAndDisplayScheduleConflict ?

Reply:
sounds good